content/posts/numpy/numpy-rng/index.md

```diff
@@ -46,7 +46,7 @@ rng = np.random.default_rng()
 rng.random()  # generate a floating point number between 0 and 1
 ```
 
-If you want to use a seed for reproducibility, [the NumPy documentation recommends](https://numpy.org/doc/stable/reference/random/index.html#quick-start) using a very large and unique number to have a different seed than anyone else and thus make sure you will have independent results. To obtain a good and unique seed you can rely on the [secrets module](https://docs.python.org/3/library/secrets.html). The following code will return a 128 bit number that you can then use as a seed:
+If you want to use a seed for reproducibility, [the NumPy documentation](https://numpy.org/doc/stable/reference/random/index.html#quick-start) recommends using a large random number, where large means at least 128 bits. The first reason for using a large random number is that this increases the probability of having a different seed than anyone else and thus independent results. The second reason is that relying only on small numbers for your seeds can lead to biases as they do not fully explore the state space of the RNG. This limitation implies that the first number generated by your RNG may not seem as random as expected due to inaccessible first internal states. For example, some numbers will never be produced as the first output. One possibility would be to pick the seed at random in the state space of the RNG but [according to Robert Kern](https://github.com/numpy/numpy/issues/25778#issuecomment-1930441151) a 128-bit random number is large enough[^1]. To generate a 128-bit random number for your seed you can rely on the [secrets module](https://docs.python.org/3/library/secrets.html):
```

Comment:
I added some details here following your question @stefanv

Reply:
I understand now; if the seed is 32-bit, it can never generate all starting values for a 128-bit space.

Reply:
Yes exactly